google/resource_compute_instance.go

```diff
@@ -715,6 +723,58 @@ func expandComputeInstance(project string, d *schema.ResourceData, config *Confi
 	}, nil
 }
 
+// return all possible Compute instances status except the one passed as parameter
+func getAllStatusBut(status string) []string {
+	allStatus := []string{
+		"PROVISIONING",
+		"REPAIRING",
+		"RUNNING",
+		"STAGING",
+		"STOPPED",
+		"STOPPING",
+		"SUSPENDED",
+		"SUSPENDING",
+		"TERMINATED",
+	}
+	for i, s := range allStatus {
+		if status == s {
+			return append(allStatus[:i], allStatus[i+1:]...)
+		}
+	}
+	return nil
+}
+
+func waitUntilInstanceHasDesiredStatus(config *Config, d *schema.ResourceData) error {
+	desiredStatus := d.Get("desired_status").(string)
```

Comment:
Is it possible here that `d.Get("desired_status")` returns `nil`? If so, I should add a check before doing the type assertion.

Reply:
`d.Get` on a field that exists in the resource schema will return the set value or the zero value for the type (`""` in this case)